cps/kobo.py

```diff
@@ -0,0 +1,544 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import base64
+import copy
+import uuid
+import os
+from datetime import datetime, tzinfo, timedelta
+from time import gmtime, strftime
+
+from b2sdk.account_info.in_memory import InMemoryAccountInfo
+from b2sdk.api import B2Api
+from jsonschema import validate, exceptions
+from flask import Blueprint, request, make_response, jsonify, json, send_file
+from flask_login import login_required
+from sqlalchemy import func
+
+from . import config, logger, kobo_auth, ub, db, helper
+from .constants import CONFIG_DIR as _CONFIG_DIR
+from .web import download_required
+
+B2_SECRETS = os.path.join(_CONFIG_DIR, "b2_secrets.json")
+
+kobo = Blueprint("kobo", __name__)
+kobo_auth.disable_failed_auth_redirect_for_blueprint(kobo)
+
+log = logger.create()
+
+
+def b64encode(data):
+    return base64.b64encode(data)
+
+
+def b64encode_json(json_data):
+    return b64encode(json.dumps(json_data))
```

Comment:
I had a problem with python3.6. I changed it to (after importing sys on the top):
```
    if sys.version_info < (3, 0):
        return b64encode(json.dumps(json_data))
    else:
        return b64encode(json.dumps(json_data).encode())
```

Reply:
Done.